xlsynth-test-helpers/src/ir_fuzz.rs

```diff
@@ -473,8 +473,12 @@ impl<'a> arbitrary::Arbitrary<'a> for FuzzSample {
         let input_bits = u.int_in_range(1..=8)?;
         // Decide how many operations to generate.
         let num_ops = u.int_in_range(0..=20)?;
-        // We always start with one available IR node: the input parameter
-        let mut available_nodes = 1;
+        // We maintain a parallel vector that records, for each existing IR node,
+        // whether it is a tuple and how many elements it contains. `None` means
+        // the node is *not* a tuple. This lets us generate in-bounds indices
+        // for `TupleIndex` operations.
+        let mut available_nodes = 1; // starts with the primary input (not a tuple)
+        let mut tuple_sizes: Vec<Option<u8>> = vec![None];
```

Comment:
Can you put a TODO to record type information for all IR nodes (which would generalize this)